File: templates/Template.ts
#c48838bd

/*
 * ============================================================
 * JAVASCRIPT SYNTAX COLOR TEST
 * ============================================================
 * Archivo diseñado para probar los diferentes tokens de
 * coloreado de sintaxis en Visual Studio Code.
 * ============================================================
 */


// ============================================================
// Comments
// ============================================================

// Comentario de una línea

/*
 * Comentario multilínea
 *
 * TODO: revisar esta función
 * FIXME: corregir este comportamiento
 * NOTE: información importante
 */


// ============================================================
// Keywords
// ============================================================

const keywordExample = true;

let variableExample = 10;

var oldVariable = 20;

if (keywordExample) {
    console.log("if");
} else {
    console.log("else");
}


// ============================================================
// Control Flow
// ============================================================

if (variableExample > 10) {
    console.log("greater");
} else if (variableExample === 10) {
    console.log("equal");
} else {
    console.log("less");
}


for (let i = 0; i < 10; i++) {
    console.log(i);
}


for (const item of ["JavaScript", "Python", "Java"]) {
    console.log(item);
}


for (const key in { name: "Manu", age: 30 }) {
    console.log(key);
}


let counter = 0;

while (counter < 5) {
    counter++;
}


do {
    counter--;
} while (counter > 0);


switch (variableExample) {
    case 10:
        console.log("Ten");
        break;

    case 20:
        console.log("Twenty");
        break;

    default:
        console.log("Other");
}


// ============================================================
// Strings
// ============================================================

const singleString = 'Hello World';

const doubleString = "Hello World";

const multilineString = "Line 1\nLine 2";

const escapedString = "He said: \"Hello\"";

const singleEscapedString = 'It\'s JavaScript';


// ============================================================
// Template Literals
// ============================================================

const name = "Manu";

const age = 30;

const templateString = `Hello ${name}`;

const templateExpression = `Age: ${age}`;

const templateCalculation = `Result: ${10 + 20}`;

const templateMethod = `Name: ${name.toUpperCase()}`;

const templateConditional = `
    Status: ${age >= 18 ? "adult" : "minor"}
`;


// ============================================================
// Template Expression Punctuation
// ============================================================

const expression = `${name}`;

const expressionCalculation = `${10 * 5 + 2}`;

const expressionProperty = `${user.name}`;

const expressionFunction = `${calculateSum(10, 20)}`;

const expressionConditional = `${
    age >= 18 ? "adult" : "minor"
}`;


// ============================================================
// Numbers
// ============================================================

const integer = 123;

const negativeNumber = -123;

const decimal = 123.456;

const scientific = 1.23e10;

const hexadecimal = 0xFF;

const binary = 0b101010;

const octal = 0o755;

const bigNumber = 12345678901234567890n;

const separatorNumber = 1_000_000;


// ============================================================
// Language Constants
// ============================================================

const trueValue = true;

const falseValue = false;

const nullValue = null;

const undefinedValue = undefined;

const nanValue = NaN;

const infinityValue = Infinity;


// ============================================================
// Escape Characters
// ============================================================

const newline = "Line 1\nLine 2";

const tab = "Column 1\tColumn 2";

const carriageReturn = "Text\r";

const backslash = "C:\\Users\\Manu";

const quote = "\"Hello\"";

const singleQuote = '\'Hello\'';

const unicode = "\u00A9";

const hexadecimalEscape = "\x41";


// ============================================================
// Function Declarations
// ============================================================

function helloWorld() {
    return "Hello World";
}


function calculateSum(a, b) {
    return a + b;
}


function processUser(
    username,
    age,
    active = true
) {
    return {
        username,
        age,
        active
    };
}


// ============================================================
// Function Parameters
// ============================================================

function parametersExample(
    requiredParameter,
    defaultParameter = 10,
    ...restParameters
) {
    return requiredParameter;
}


function destructuringParameters(
    { name, age },
    [first, second]
) {
    return {
        name,
        age,
        first,
        second
    };
}


// ============================================================
// Function Calls
// ============================================================

helloWorld();

calculateSum(10, 20);

processUser(
    "Manu",
    30
);

console.log("Hello World");

Math.max(10, 20, 30);

JSON.stringify({
    name: "Manu"
});


// ============================================================
// Arrow Functions
// ============================================================

const add = (a, b) => {
    return a + b;
};


const multiply = (a, b) => a * b;


const greet = name => `Hello ${name}`;


const getUser = () => ({
    id: 1,
    name: "Manu"
});


// ============================================================
// Classes & Types
// ============================================================

class User {

    constructor(id, username) {
        this.id = id;
        this.username = username;
    }

    getUsername() {
        return this.username;
    }

    toString() {
        return `User: ${this.username}`;
    }
}


// ============================================================
// Class Inheritance
// ============================================================

class Admin extends User {

    constructor(id, username, permissions) {

        super(id, username);

        this.permissions = permissions;
    }

    hasPermission(permission) {
        return this.permissions.includes(permission);
    }
}


// ============================================================
// Static Members
// ============================================================

class Configuration {

    static VERSION = "1.0.0";

    static MAX_USERS = 100;

    static getVersion() {
        return Configuration.VERSION;
    }
}


// ============================================================
// Variables
// ============================================================

let username = "Manu";

let currentAge = 30;

let active = true;

let data = null;

let result;

let object = {};

let array = [];


// ============================================================
// Constants
// ============================================================

const MAX_VALUE = 100;

const MIN_VALUE = 0;

const API_VERSION = "v1";

const DEFAULT_TIMEOUT = 3000;

const APPLICATION_NAME = "Color Test";


// ============================================================
// Object Properties
// ============================================================

const user = {
    id: 1,
    name: "Manu",
    age: 30,
    active: true
};


user.name;

user.age;

user.active;

user["name"];


// ============================================================
// Object Keys
// ============================================================

const objectWithKeys = {
    id: 1,
    name: "Manu",
    email: "manu@example.com",
    active: true,

    address: {
        street: "Main Street",
        city: "Madrid",
        country: "Spain"
    }
};


// ============================================================
// Computed Object Keys
// ============================================================

const propertyName = "username";

const computedObject = {
    [propertyName]: "Manu",

    [`user_${propertyName}`]: "value"
};


// ============================================================
// Destructuring
// ============================================================

const {
    id,
    name: userName,
    age: userAge
} = user;


const [first, second, third] = [
    "one",
    "two",
    "three"
];


// ============================================================
// Operators
// ============================================================

const addition = 10 + 5;

const subtraction = 10 - 5;

const multiplication = 10 * 5;

const division = 10 / 5;

const remainder = 10 % 3;

const exponentiation = 2 ** 3;


const equal = 10 == "10";

const strictEqual = 10 === 10;

const notEqual = 10 != 5;

const strictNotEqual = 10 !== 5;

const greater = 10 > 5;

const less = 5 < 10;

const greaterEqual = 10 >= 10;

const lessEqual = 5 <= 10;


const logicalAnd = true && true;

const logicalOr = true || false;

const logicalNot = !true;


const nullish = null ?? "default";

const optional = user?.address?.city;


// ============================================================
// Assignment Operators
// ============================================================

let value = 10;

value += 5;

value -= 2;

value *= 3;

value /= 2;

value %= 3;

value **= 2;

value ??= 100;

value ||= 50;

value &&= 20;


// ============================================================
// Increment / Decrement
// ============================================================

let number = 0;

number++;

number--;

++number;

--number;


// ============================================================
// Ternary Operator
// ============================================================

const status = active
    ? "ACTIVE"
    : "INACTIVE";


// ============================================================
// Punctuation & Brackets
// ============================================================

const punctuation = {
    array: [
        1,
        2,
        3
    ],

    object: {
        key: "value"
    },

    function: calculateSum(
        10,
        20
    )
};


// ============================================================
// Arrays
// ============================================================

const languages = [
    "JavaScript",
    "Python",
    "Java",
    "TypeScript"
];


languages.push("C#");

languages.pop();

languages.map(
    language => language.toUpperCase()
);

languages.filter(
    language => language.length > 5
);

languages.forEach(
    language => console.log(language)
);


// ============================================================
// Spread Operator
// ============================================================

const numbers = [1, 2, 3];

const moreNumbers = [
    ...numbers,
    4,
    5
];


const originalUser = {
    name: "Manu",
    age: 30
};


const copiedUser = {
    ...originalUser,
    active: true
};


// ============================================================
// Rest Operator
// ============================================================

function sumAll(first, ...numbers) {

    return numbers.reduce(
        (total, number) => total + number,
        first
    );
}


// ============================================================
// Regular Expressions
// ============================================================

const regex = /^[a-zA-Z0-9_]+$/;

const emailRegex =
    /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

const match = "Manu123".match(regex);

const validEmail =
    "manu@example.com".match(emailRegex);


// ============================================================
// RegExp Constructor
// ============================================================

const dynamicRegex = new RegExp(
    "^[a-zA-Z0-9]+$"
);


// ============================================================
// Error Handling
// ============================================================

try {

    const parsed = JSON.parse(
        '{"name":"Manu"}'
    );

    console.log(parsed);

} catch (error) {

    console.error(error);

} finally {

    console.log("Finished");
}


// ============================================================
// Throw
// ============================================================

function validateAge(age) {

    if (age < 0) {

        throw new Error(
            "Age cannot be negative"
        );
    }

    return true;
}


// ============================================================
// Async / Await
// ============================================================

async function fetchData() {

    const response = await fetch(
        "https://example.com/api"
    );

    const data = await response.json();

    return data;
}


async function loadData() {

    try {

        const data = await fetchData();

        console.log(data);

    } catch (error) {

        console.error(error);
    }
}


// ============================================================
// Promise
// ============================================================

const promise = new Promise(
    (resolve, reject) => {

        const success = true;

        if (success) {
            resolve("Success");
        } else {
            reject(
                new Error("Failed")
            );
        }
    }
);


promise
    .then(result => {
        console.log(result);
    })
    .catch(error => {
        console.error(error);
    })
    .finally(() => {
        console.log("Finished");
    });


// ============================================================
// Modules
// ============================================================

import fs from "fs";

import path from "path";

import {
    readFile,
    writeFile
} from "fs/promises";


// ============================================================
// Export
// ============================================================

export function exportedFunction() {

    return "Exported";
}


export const exportedConstant = 100;


export default class ExportedClass {

    run() {
        return true;
    }
}


// ============================================================
// Namespaces & Modules
// ============================================================

// JavaScript no tiene namespaces clásicos como C#.
// Los módulos ES (`import` / `export`) cumplen una
// función equivalente de organización.


const modulePath = path.join(
    "C:",
    "Users",
    "Manu"
);


// ============================================================
// Decorators & Annotations
// ============================================================

// Los decorators dependen de la versión/configuración
// de JavaScript y del runtime/transpilador.
//
// @decorator
// class Example {}


// ============================================================
// Getter / Setter
// ============================================================

class Person {

    constructor(name) {
        this._name = name;
    }

    get name() {
        return this._name;
    }

    set name(value) {
        this._name = value;
    }
}


// ============================================================
// Private Properties
// ============================================================

class BankAccount {

    #balance = 0;

    deposit(amount) {

        this.#balance += amount;
    }

    getBalance() {

        return this.#balance;
    }
}


// ============================================================
// Symbols
// ============================================================

const symbol = Symbol("id");

const objectWithSymbol = {

    [symbol]: 123
};


// ============================================================
// Map / Set
// ============================================================

const map = new Map();

map.set("name", "Manu");

map.set("age", 30);


const set = new Set([
    "JavaScript",
    "Python",
    "Java"
]);


// ============================================================
// BigInt
// ============================================================

const bigInteger = 123456789012345678901234567890n;

const anotherBigInteger = BigInt(
    "12345678901234567890"
);


// ============================================================
// Optional Chaining
// ============================================================

const city =
    user?.address?.city;

const methodResult =
    user?.getUsername?.();


// ============================================================
// Nullish Coalescing
// ============================================================

const displayName =
    userName ?? "Unknown";


// ============================================================
// Invalid / Deprecated
// ============================================================

// Código inválido intencionadamente comentado:
//
// const invalid =
//
// function invalidFunction(
//
// if true
//
// const x = {;
//
//
// Deprecated:
//
// escape("hello");
//
// unescape("hello");


// ============================================================
// HTML / XML Tags
// ============================================================

// Estas etiquetas NO forman parte de JavaScript puro.
// Se incluyen únicamente para probar si el tema/gramática
// está mezclando HTML con JavaScript.

/*
<div class="container">
    <span id="title">Hello</span>
</div>
*/


// ============================================================
// Attributes
// ============================================================

// Tampoco son elementos propios de JavaScript.
//
// class="container"
// id="title"
// data-value="123"


// ============================================================
// CSS Properties
// ============================================================

// No son tokens propios de JavaScript.
//
// color: red;
// background-color: blue;
// font-size: 16px;


// ============================================================
// JSX / TSX Components
// ============================================================

// JSX no es JavaScript estándar, pero es habitual en
// archivos .jsx / .tsx.

/*
const component = (
    <User
        name="Manu"
        age={30}
    />
);
*/


// ============================================================
// Main
// ============================================================

function main() {

    const testUser = new User(
        1,
        "Manu"
    );

    console.log(
        testUser.getUsername()
    );

    console.log(
        calculateSum(
            10,
            20
        )
    );
}


main();

File: templates/Template.py
"""
============================================================
PYTHON SYNTAX COLOR TEST
============================================================
Archivo diseñado para probar los diferentes tokens de
coloreado de sintaxis en Visual Studio Code.
============================================================
"""

# ============================================================
# Comments
# ============================================================

# Comentario de una línea

"""
Comentario multilínea.
También debería tener un color diferente.
"""

# TODO: comentario con TODO
# FIXME: comentario con FIXME
# NOTE: comentario con NOTE


# ============================================================
# Keywords
# ============================================================

import os
import sys
from pathlib import Path
from typing import List, Dict, Optional, TypeVar, Generic


# ============================================================
# Strings
# ============================================================

single_string = 'Hello World'

double_string = "Hello World"

multiline_string = """
This is a multiline string.
It contains several lines.
"""

raw_string = r"C:\Users\Manu\Documents\file.txt"

byte_string = b"Binary data"


# ============================================================
# Template Literals
# ============================================================
# Python no tiene Template Literals como JavaScript.
# Python utiliza f-strings.


name = "Manu"
age = 30

f_string = f"Hello {name}"

f_string_expression = f"Name: {name}, Age: {age}"

f_string_calculation = f"Result: {10 + 20}"

f_string_format = f"{age:03d}"


# ============================================================
# Template Expression Punctuation
# ============================================================
# En Python, las expresiones dentro de f-strings usan {}.

message = f"Hello {name.upper()}"

calculation = f"Result: {10 * 5 + 2}"

conditional = f"Status: {'active' if age >= 18 else 'inactive'}"


# ============================================================
# Numbers
# ============================================================

integer = 123

negative_integer = -123

float_number = 123.456

scientific_number = 1.23e10

hexadecimal = 0xFF

binary = 0b101010

octal = 0o755

complex_number = 3 + 4j

large_number = 1_000_000


# ============================================================
# Language Constants
# ============================================================

true_value = True

false_value = False

none_value = None

ellipsis_value = ...


# ============================================================
# Escape Characters
# ============================================================

newline = "Line 1\nLine 2"

tab = "Column 1\tColumn 2"

carriage_return = "Text\r"

backslash = "C:\\Users\\Manu"

quotes = "He said: \"Hello\""

single_quotes = 'It\'s Python'

unicode_character = "\u00A9"

hex_character = "\x41"


# ============================================================
# Variables
# ============================================================

name = "Manu"

counter = 10

result = 25.5

active = True

data = None


# ============================================================
# Constants
# ============================================================

MAX_VALUE = 100

MIN_VALUE = 0

API_VERSION = "v1"

DEFAULT_TIMEOUT = 30


# ============================================================
# Function Declarations
# ============================================================

def hello_world():
    """Simple function."""
    return "Hello World"


def calculate_sum(a, b):
    """Calculate the sum of two numbers."""
    return a + b


def process_user(
    username: str,
    age: int,
    active: bool = True
) -> str:
    """Process a user."""
    return f"{username}: {age}"


# ============================================================
# Function Parameters
# ============================================================

def parameters_example(
    required_parameter,
    typed_parameter: str,
    default_parameter: int = 10,
    *args,
    **kwargs
):
    return required_parameter


# ============================================================
# Function Calls
# ============================================================

hello_world()

calculate_sum(10, 20)

process_user(
    username="Manu",
    age=30
)

print("Hello World")


# ============================================================
# Control Flow
# ============================================================

if age >= 18:
    status = "adult"
elif age >= 13:
    status = "teenager"
else:
    status = "child"


# ============================================================
# For Loop
# ============================================================

for i in range(10):
    print(i)


for item in ["Python", "Java", "TypeScript"]:
    print(item)


# ============================================================
# While Loop
# ============================================================

counter = 0

while counter < 10:
    counter += 1


# ============================================================
# Break / Continue / Pass
# ============================================================

for number in range(20):

    if number == 5:
        continue

    if number == 15:
        break

    print(number)


def not_implemented():
    pass


# ============================================================
# Match / Case
# ============================================================

status = "active"

match status:

    case "active":
        print("User is active")

    case "inactive":
        print("User is inactive")

    case "pending":
        print("User is pending")

    case _:
        print("Unknown status")


# ============================================================
# Classes & Types
# ============================================================

class User:

    def __init__(
        self,
        user_id: int,
        username: str
    ):
        self.user_id = user_id
        self.username = username

    def get_username(self) -> str:
        return self.username

    def __str__(self) -> str:
        return self.username


# ============================================================
# Class Inheritance
# ============================================================

class Admin(User):

    def __init__(
        self,
        user_id: int,
        username: str,
        permissions: list[str]
    ):
        super().__init__(user_id, username)

        self.permissions = permissions

    def has_permission(
        self,
        permission: str
    ) -> bool:
        return permission in self.permissions


# ============================================================
# Interfaces & Type Parameters
# ============================================================
# Python no tiene interfaces como Java/TypeScript.
# Se utilizan normalmente ABC o Protocol.


from abc import ABC, abstractmethod


class Repository(ABC):

    @abstractmethod
    def find_by_id(self, user_id: int):
        pass

    @abstractmethod
    def find_all(self):
        pass


# ============================================================
# Generic / Type Parameters
# ============================================================

T = TypeVar("T")


class Container(Generic[T]):

    def __init__(self, value: T):
        self.value = value

    def get(self) -> T:
        return self.value


string_container = Container[str]("Hello")

integer_container = Container[int](123)


# ============================================================
# Type Annotations
# ============================================================

username: str = "Manu"

age: int = 30

price: float = 19.99

enabled: bool = True

user: Optional[User] = None

names: List[str] = []

users: Dict[str, User] = {}


# ============================================================
# Object Properties
# ============================================================

user = User(
    user_id=1,
    username="Manu"
)

user.username

user.user_id

user.get_username()


# ============================================================
# Object Keys
# ============================================================
# En Python, los diccionarios utilizan claves.


user_data = {
    "id": 1,
    "username": "Manu",
    "email": "manu@example.com",
    "active": True
}


# ============================================================
# Dictionary Access
# ============================================================

user_data["username"]

user_data["email"]

user_data.get("active")


# ============================================================
# Enum Members
# ============================================================

from enum import Enum


class Status(Enum):

    ACTIVE = "active"

    INACTIVE = "inactive"

    PENDING = "pending"


current_status = Status.ACTIVE

if current_status == Status.ACTIVE:
    print("Active")


# ============================================================
# Operators
# ============================================================

a = 10

b = 5

addition = a + b

subtraction = a - b

multiplication = a * b

division = a / b

floor_division = a // b

modulo = a % b

power = a ** b

comparison = a == b

not_equal = a != b

greater = a > b

less = a < b

greater_equal = a >= b

less_equal = a <= b

logical_and = a > 0 and b > 0

logical_or = a > 0 or b > 0

logical_not = not enabled

membership = "Python" in ["Python", "Java"]

identity = a is b


# ============================================================
# Assignment Operators
# ============================================================

value = 10

value += 5

value -= 2

value *= 3

value /= 2

value //= 2

value %= 3

value **= 2


# ============================================================
# Punctuation & Brackets
# ============================================================

items = [
    "one",
    "two",
    "three"
]

data = {
    "name": "Manu",
    "age": 30
}

coordinates = (
    10,
    20
)

result = calculate_sum(
    10,
    20
)


# ============================================================
# List / Set / Dictionary Comprehensions
# ============================================================

numbers = [1, 2, 3, 4, 5]

squares = [
    number ** 2
    for number in numbers
]

even_numbers = [
    number
    for number in numbers
    if number % 2 == 0
]

unique_values = {
    number
    for number in numbers
}

number_map = {
    number: number ** 2
    for number in numbers
}


# ============================================================
# Lambda
# ============================================================

double = lambda value: value * 2

result = double(10)


# ============================================================
# Decorators & Annotations
# ============================================================

@property
def username_property(self):
    return self.username


@staticmethod
def static_method():
    return "Static"


@classmethod
def class_method(cls):
    return cls()


# ============================================================
# Import / Modules
# ============================================================

import json

import datetime

from pathlib import Path

from collections import defaultdict

from typing import Optional


# ============================================================
# Namespaces & Modules
# ============================================================

os.path.join(
    "C:",
    "Users",
    "Manu"
)

datetime.datetime.now()

Path("example.txt").exists()


# ============================================================
# Exceptions
# ============================================================

try:

    result = 10 / 0

except ZeroDivisionError as error:

    print(error)

except Exception as error:

    print(error)

else:

    print("No error")

finally:

    print("Finished")


# ============================================================
# Raise
# ============================================================

if age < 0:

    raise ValueError(
        "Age cannot be negative"
    )


# ============================================================
# Context Managers
# ============================================================

with open(
    "example.txt",
    "r",
    encoding="utf-8"
) as file:

    content = file.read()


# ============================================================
# Async / Await
# ============================================================

import asyncio


async def fetch_data():

    await asyncio.sleep(1)

    return "Data"


async def main():

    data = await fetch_data()

    print(data)


# ============================================================
# Regular Expressions
# ============================================================

import re


pattern = r"^[a-zA-Z0-9_]+$"

regex = re.compile(pattern)

match = regex.match("Python_123")


# ============================================================
# Invalid / Deprecated
# ============================================================

# Invalid syntax examples are intentionally commented out
# so that the file remains executable.

# invalid =

# if True

# def invalid_function(

# Deprecated example:

# imp module is deprecated in modern Python versions.
# import imp


# ============================================================
# Special / Dunder Methods
# ============================================================

class Example:

    def __init__(self):
        self.value = 10

    def __str__(self):
        return str(self.value)

    def __repr__(self):
        return f"Example(value={self.value})"

    def __len__(self):
        return 1

    def __eq__(self, other):
        return self.value == other.value


# ============================================================
# Main Entry Point
# ============================================================

if __name__ == "__main__":

    user = User(
        user_id=1,
        username="Manu"
    )

    print(user)

    print(
        calculate_sum(
            10,
            20
        )
    )